shared/events.js

```diff
@@ -348,16 +362,50 @@ export class PostMessage extends HashableAction {
   static fields = PostMessage.Fields([
     ['from', 'string'],
     ['headers', 'string', 'repeated'],
-    ['message', 'string']
+    ['message', 'string'],
+    ['sentDate', 'string'],
+    ['read', 'bool']
   ])
+  constructor (data: JSONObject, parentHash?: string) {
+    super({...data, read: false}, parentHash)
+  }
 }
-
 export class PostInvite extends HashableAction {
   static fields = PostInvite.Fields([
-    ['groupId', 'string']
+    ['groupId', 'string'],
+    ['sentDate', 'string'],
+    ['read', 'bool']
+  ])
+  constructor (data: JSONObject, parentHash?: string) {
+    super({...data, read: false}, parentHash)
+  }
+}
+export class RecordInvitation extends HashableAction {
+  static fields = RecordInvitation.Fields([
+    ['username', 'string'],
+    ['inviteHash', 'string'],
+    ['sentDate', 'string']
+  ])
+}
+export class AcceptInvitation extends HashableAction {
+  static fields = AcceptInvitation.Fields([
+    ['username', 'string'],
+    ['inviteHash', 'string'],
+    ['acceptanceDate', 'string']
+  ])
+}
+export class DeclineInvitation extends HashableAction {
+  static fields = DeclineInvitation.Fields([
+    ['username', 'string'],
+    ['inviteHash', 'string'],
+    ['declinedDate', 'string']
+  ])
+}
+export class AcknowledgeFounder extends HashableAction {
+  static fields = AcknowledgeFounder.Fields([
+    ['username', 'string']
```

Comment:
This appears to be a GroupContract related action, but it's in the MailboxContract section. Needs to be moved, along with the other invitation related actions and anything else that's mislocated.